syntax = "proto3";
package client;

option go_package = "../proto";

service FileService {
    rpc Upload(FileUploadRequest) returns (stream FileChunk);
    rpc Download(stream FileChunk) returns (Response);
    // rpc Delete(FileDeleteRequest) returns (FileDeleteResponse)
}

message FileUploadRequest {
    int32 request_id = 1;
    string file_name = 2;
}

message FileChunk {
    int32 request_id = 1;
    string file_name = 2;
    bytes chunk = 3;
}

message Response {
    int32 request_id = 1;
    bool success = 2;
}